app/models/aggregated_cash_income.rb

```diff
@@ -0,0 +1,106 @@
+class AggregatedCashIncome
+  include ActiveModel::Model
+
+  CATEGORIES= %i[
+      benefits
+      friends_or_family
+      maintenance_in
+      property_or_lodger
+      pension
+  ].freeze
+
+  RANGE = (1..3).freeze
+
+  def self.category_attributes(categories)
+    categories.each do |category|
+      attr_accessor "check_box_#{category}".to_sym
+      RANGE.each do |i|
+        attr_accessor "#{category}_month_#{i}".to_sym
+      end
+    end
+  end
+
+  attr_accessor :legal_aid_application_id,
+                :category,
+                :none_selected
+
+  category_attributes CATEGORIES
+
+  validate :check_category
+  validate :anything_selected?
+
+  def checkbox_integrity(params)
+    CATEGORIES.each do |category|
+      checkbox_selected = params["check_box_#{category}"] == 'true'
+
+      if checkbox_selected
+        @category = category
+        check_category
+      else
+        args = search_args(category)
+        CashTransaction.where(**args).destroy_all
+      end
+    end
+
+    anything_selected?
+  end
+
+  def save(params)
+    params.each do |key, value|
+      __send__("#{key}=",value) if value != ''
+    end
+
+    checkbox_integrity(params)
+
+    return false unless errors.empty?
+
+    save_cash_transaction_records
+  end
+
+  private
+
+  def save_cash_transaction_records
+    CATEGORIES.each do |category|
+      save_category(category) if checkbox_for(category) == 'true'
+    end
+  end
+
+  def save_category(category)
+    RANGE.each do |i|
+      args = search_args(category)
+      amount = __send__("#{category}_month_#{i}".to_sym)
+      date = calculated_date(i)
+
+      CashTransaction.where(**args, transaction_date: date).destroy_all
+      CashTransaction.create!(**args, transaction_date: date, amount: amount)
+    end
+  end
+
+  def anything_selected?
+    errors.add(:none_selected, :blank) unless category
+  end
+
+  def check_category
+    RANGE.each do |i|
+      method = "#{category}_month_#{i}".to_sym
+      value = __send__(method)
+      errors.add(method, :blank) unless value.present?
+    end
+  end
+
+  def search_args(category)
+    return {
+      legal_aid_application_id: legal_aid_application_id,
+      transaction_type_id: TransactionType.credits.find_by(name: category).id
+    }
+  end
+
+  def calculated_date(month_number)
+    date = Date.today.at_beginning_of_month - (month_number).months
+    date.strftime('%-d %B %Y')
```

Comment:
we can save a `Date` object to the database, so we don't have to convert it into a string here.

Reply:
So we did this because without explicitly setting it with strftime, the date is formatted as ` 1 November 2020` (with a prefixed space). I wasn't sure if there was a default helper doing this when it is saved to the DB, but by doing this, when we query records using the date as a parameter when doing `CashTransactions.where(...etc` it meant we could reassured there was consistency in it being stored and when being queried. Again, it may be we might have missed something so we can remove this, but currently if it does then we have to convert it like this anyway when querying.